File: scripts/csv_convert.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"io/ioutil"
	"log"
	"os"
)

func main() {
	input := "../data/items-with-bio.json"
	output := "../data/items-with-bio.csv"

	outFile, err := os.Create(output)
	if err != nil {
		log.Fatal(err)
	}
	w := csv.NewWriter(outFile)

	data, err := ioutil.ReadFile(input)
	if err != nil {
		log.Fatal(err)
	}

	var documents []map[string]interface{}
	err = json.Unmarshal(data, &documents)
	if err != nil {
		log.Fatal(err)
	}

	nFields := len(documents[0])

	// Get list of map keys from first map in slice
	headers := make([]string, 0, nFields)
	for header := range documents[0] {
		headers = append(headers, header)
	}

	w.Write(headers)

	row := make([]string, nFields)
	for _, document := range documents {
		for i, header := range headers {
			field, err := json.Marshal(document[header])
			if err != nil {
				log.Fatal(err)
			}
			row[i] = string(field)
		}
		w.Write(row)
	}

	log.Println("Number of documents", len(documents))
}

File: scripts/csv_convert.py
#!/usr/bin/env python
# ##### -*- coding: UTF8 -*-
"""
Convert JSON to CSV
~~~~~~~~~~~~~~~~~~~
"""
import sys

usage = """
csv_convert.py data.json data.csv

data.json contains an array of identical objects.

"""

import json
import requests

if __name__ == '__main__':

    if len(sys.argv) < 3:
        print(usage)
        sys.exit(1)

    data, outfile = sys.argv[1:]

    with open(data, 'r') as fo:
        documents = json.load(fo, encoding='ascii')

    # Establish headers
    ref = documents[0]
    headers = ref.keys()

    with open(outfile, 'w') as fo:
        fo.writelines(",".join(headers) + "\n")

        for i, document in enumerate(documents):
            fo.write(",".join(str(v) for v in document.values()) + "\n")
